app/controllers/publishers/site_banners_controller.rb

```diff
@@ -5,14 +5,104 @@ class Publishers::SiteBannersController < ApplicationController
 
   MAX_IMAGE_SIZE = 10_000_000
 
-  def new
-    @site_banner = current_publisher.site_banner || SiteBanner.new
+  #Fetch Banner by channel_id, if it does not exist create a template banner.
+  def fetch
+    site_banner = current_publisher.site_banners.where(channel_id: params[:channel_id]).first
+    if site_banner.present?
+      data = site_banner.read_only_react_property
+      data[:backgroundImage] = data[:backgroundUrl]
+      data[:logoImage] = data[:logoUrl]
+      render(json: data.to_json)
+    else
+      site_banner = SiteBanner.new
+      site_banner.update(
+        publisher_id: current_publisher.id,
+        channel_id: params[:channel_id],
+        title: 'Brave Rewards',
+        description:
+'Thanks for stopping by. We joined Brave\'s vision of protecting your privacy because we believe that fans like you would support us in our effort to keep the web a clean and safe place to be.
+
+Your tip is much appreciated and it encourages us to continue to improve our content.',
+        donation_amounts: [1, 5, 10],
+        default_donation: 5,
+        social_links: {'youtube': '', 'twitter': '', 'twitch': ''}
+      )
+      data = site_banner.read_only_react_property
+      data[:backgroundImage] = nil
+      data[:logoImage] = nil
+      render(json: data.to_json)
+    end
   end
 
-  def create
-    site_banner = current_publisher.site_banner || SiteBanner.new
+  def channels
+
+    # Handle edge case where user hasn't created any channels yet
+    if current_publisher.channels.length == 0
+      data = {}
+      data[:channels] = [{'id' => '00000000-0000-0000-0000-000000000000', 'name' => ' ', 'type' => 'website'}]
+      data[:channel_mode] = false
+      data[:channel_index] = 0
+      render(json: data.to_json) and return
+    end
+
+    if current_publisher.site_banners.where(channel_id: '00000000-0000-0000-0000-000000000000').exists?
+      current_publisher.site_banners.where(channel_id: '00000000-0000-0000-0000-000000000000').first.delete
+    end
+
+    channels = []
+
+    current_publisher.channels.each do |c|
+
+    case c.details_type
+
+      when "SiteChannelDetails"
+        channel_details = SiteChannelDetails.find_by_id(c.details_id)
+        channel_type = "website"
+        channel_name = channel_details.url
+      when "YoutubeChannelDetails"
+        channel_details = YoutubeChannelDetails.find_by_id(c.details_id)
+        channel_type = "youtube"
+        channel_name = channel_details.auth_name
+      when "TwitchChannelDetails"
+        channel_details = TwitchChannelDetails.find_by_id(c.details_id)
+        channel_type = "twitch"
+        channel_name = channel_details.name
+
+    end
+
+    channel = {
+              'id' => c.id,
+              'name' => channel_name,
+              'type' => channel_type
+              }
+
+    channels.push(channel)
+    end
+
+    channel_mode = true
```

Comment:
I was thinking this  block could potentially be more ruby-esque so we don't have define these temp variables

```ruby
    default_banner = current_publisher.site_banners.find_by(default: true)
    if default_banner.present?
      default_id = default_banner.channel_id
      channel_index = channels.index { |c| c["id"] == default_id }
    end

    data = {}
    data[:channels] = channels
    data[:channel_mode] = default_banner.empty?
    data[:channel_index] = channel_index || 0
    render json: data.to_json
```

Reply:
Done ✔️